lib/strings.nix

```
@@ -205,7 +205,12 @@ rec {
        normalizePath "/a//b///c/"
        => "/a/b/c/"
   */
-  normalizePath = s: (builtins.foldl' (x: y: if y == "/" && hasSuffix "/" x then x else x+y) "" (stringToCharacters s));
+  normalizePath = s:
+    if isPath s then throw ''
+      lib.strings.normalizePath: The argument (${toString s}) is not a string but a path, which is not supported.
+          Values of the path data type are always normalised in Nix, there's no need to do that manually.
+          To normalise subpaths when working with path data types, please use `lib.path.subpath.normalise`.''
+    else builtins.foldl' (x: y: if y == "/" && hasSuffix "/" x then x else x+y) "" (stringToCharacters s);
```

```suggestion
    else
      builtins.foldl'
        (x: y: if y == "/" && hasSuffix "/" x then x else x+y)
        ""
        (stringToCharacters s);
```

Now we can read it, yay!